### PR TITLE
feat: caching bodies to avoid unnecessary object creation

### PR DESCRIPTION
**Pull request by RobPasMue**
## Description
Allow for body caching if no new bodies have been added. This reduces the amount of time we spend on object creation...

## Issue linked
None

## Checklist
- [X] I have tested my changes locally.
- [X] I have added necessary documentation or updated existing documentation.
- [X] I have followed the coding style guidelines of this project.
- [X] I have added appropriate unit tests.
- [X] I have reviewed my changes before submitting this pull request.
- [X] I have linked the issue or issues that are solved to the PR if any.
- [X] I have assigned this PR to myself.
- [X] I have added the minimum version decorator to any new backend method implemented.
- [X] I have made sure that the title of my PR follows [Conventional commits style](https://www.conventionalcommits.org/en/v1.0.0/#summary) (e.g. ``feat: extrude circle to cylinder``)

### CODE REVIEW

File: tests/integration/test_design.py [hunk 1]
```diff
@@ -2738,3 +2738,49 @@ def test_surface_body_creation(modeler: Modeler):
...
+        sketch=sketch_cylinder,
+        distance=Distance(60, unit=UNITS.m),
+    )
+
+    my_bodies = design.bodies
+    my_bodies_2 = design.bodies
+
+    # We should make sure that the object memory addresses are the same
+    for body1, body2 in zip(my_bodies, my_bodies_2):
+        assert body1 is body2  # We are comparing the memory addresses
```
Comment (MaxJPRey) [inline tests/integration/test_design.py:2773]: I like this way to test it. I don't actually see any other way to do it.

Comment (RobPasMue) [reply tests/integration/test_design.py:2773]: Correct... Testing this has to be done at "object" comparison level. They need to be exactly the same.

**Comment by RobPasMue** [comment]
Needs investigation from my side... let's see what's going on.

**Comment by RobPasMue** [comment]
Okay so, on body deletion we also need to clear the cache. That was missing - my bad. Ready to go now!

**Pull request merged**